src/persist-client/src/internal/machine.rs

```diff
@@ -1811,7 +1708,14 @@ pub mod datadriven {
         let now = (datadriven.client.cfg.now)();
         let (_, maintenance) = datadriven
             .machine
-            .compare_and_append_idempotent(&batch, &writer_id, now, &token, indeterminate)
+            .compare_and_append_idempotent(
+                &batch,
+                &writer_id,
+                now,
+                &token,
+                &HandleDebugState::default(),
+                indeterminate,
+            )
```

Comment:
This code is covered by unit tests but not by other tests at higher levels. Any chances of covering it?
I am also happy to support if needed.

Reply:
This entire `datadriven` module is effectively support code for unit tests... and it's marked `#[cfg(test)]`, so it's not compiled into the main materialize binary, and it's impossible for a higher-level test to invoke it right now. (Similar for the other test-helper method you flagged.) We do have good non-unit-test coverage of the underlying `compare_and_append_idempotent` method that this test helper method also covers, though!

Does that sound reasonable, or are there other ways you'd suggest covering unit-test-support code like this?

Reply:
Right. I missed the `#[cfg(test)]` marker.
Fine with me, thanks for the explanation!

Reply:
Thank you for checking!